capstone/src/main/java/com/google/step/search/ElasticsearchClient.java

```diff
@@ -0,0 +1,102 @@
+// limitations under the License.
+
+package com.google.step.search;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpContent;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.Json;
+import com.google.appengine.repackaged.com.google.gson.Gson;
+import com.google.step.data.RestaurantHeader;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * The {@code ElasticsearchClient} is a class that sends requests to add {@code RestaurantHeaders}
+ * to the "restaurants" index of Elasticsearch.
+ */
+public class ElasticsearchClient {
+
+  private static final String elasticsearchHostname = "10.128.0.2";
+  private static final short elasticsearchPort = 9200;
+  private final String elasticsearchUriString;
+
+  private static final String RESTAURANTS = "restaurants";
+
+  private final HttpRequestFactory requestFactory;
+
+  Gson gson = new Gson();
+
+  ElasticsearchClient(HttpTransport transport) {
+    requestFactory = transport.createRequestFactory();
+
+    URIBuilder uriBuilder = new URIBuilder()
+        .setScheme("http")
+        .setHost(elasticsearchHostname)
+        .setPort(elasticsearchPort);
+    elasticsearchUriString = uriBuilder.toString();
+  }
+
+  public ElasticsearchClient() {
+    this(new NetHttpTransport());
+  }
+
+  /**
+   * Given a {@code RestaurantHeader}, sends HTTP request to Elasticsearch server to add a document
+   * to the "restaurants" index representing a {@code restaurantHeader}. If the request fails for
+   * reason, an HTTP status code of 500 will be returned.
+   * @param restaurantHeader a {@code RestaurantHeader}
+   * @return integer representing the HTTP status code of the request
+   */
+  public int updateRestaurantHeader(RestaurantHeader restaurantHeader) {
+    String restaurantKey = String.valueOf(restaurantHeader.getRestaurantKey());
+
+    GenericUrl requestUrl = new GenericUrl(elasticsearchUriString);
+    requestUrl.setPathParts(Arrays.asList("", RESTAURANTS, "_doc", restaurantKey));
+
+    String requestBody = gson.toJson(restaurantHeader);
+    HttpContent putRequestContent = new ByteArrayContent(Json.MEDIA_TYPE,
+        requestBody.getBytes(StandardCharsets.UTF_8));
+
+    int statusCode;
+    try {
+      HttpRequest request = requestFactory.buildPutRequest(requestUrl, putRequestContent);
+      HttpResponse response = request.execute();
+
+      statusCode = response.getStatusCode();
+    } catch (IOException e) {
+      statusCode = HttpURLConnection.HTTP_INTERNAL_ERROR;
+    }
+
+    return statusCode;
```

Comment:
Final error-related comment, and probably the one you should really be doing: Consider just throwing HttpResponseException, IOException, and letting the client handle it as they see fit. Just a status code isn't really enough information, so passing up the full info is best.